dds/DCPS/Discovery.cpp

```diff
@@ -68,6 +68,43 @@ Discovery::create_bit_topics(DomainParticipantImpl* participant)
                      DDS::RETCODE_ERROR);
   }
 
+  // Participant location topic
+  type_support =
+    Registered_Data_Types->lookup(participant, BUILT_IN_PARTICIPANT_LOCATION_TOPIC_TYPE);
+
+  if (CORBA::is_nil(type_support)) {
+    DDS::ParticipantLocationBuiltinTopicDataTypeSupport_var ts =
+      new DDS::ParticipantLocationBuiltinTopicDataTypeSupportImpl;
+
+    DDS::ReturnCode_t ret = ts->register_type(participant,
+      BUILT_IN_PARTICIPANT_LOCATION_TOPIC_TYPE);
+
+    if (ret != DDS::RETCODE_OK) {
+      ACE_ERROR_RETURN((LM_ERROR,
+        ACE_TEXT("(%P|%t) ")
+        ACE_TEXT("Discovery::create_bit_topics, ")
+        ACE_TEXT("register BUILT_IN_PARTICIPANT_LOCATION_TOPIC_TYPE returned %d.\n"),
+        ret),
+        ret);
```

Comment:
This isn't critical, but there is a function called `retcode_to_string` in `SafetyProfileStreams.h` that stringifies DDS return codes.